pandas/core/arrays/period.py

```diff
@@ -110,25 +325,10 @@ def freq(self, value):
         msg = ('Setting {cls}.freq has been deprecated and will be '
                'removed in a future version; use {cls}.asfreq instead. '
                'The {cls}.freq setter is not guaranteed to work.')
-        warnings.warn(msg.format(cls=type(self).__name__),
-                      FutureWarning, stacklevel=2)
+        warnings.warn(msg.format(cls='PeriodIndex'),
```

Comment:
Why this change?

Reply:
What do you mean exactly? 
About what to do with the actual wording the warning message? (PeriodIndex vs PeriodArray)

Reply:
Exactly. If PeriodArray isn't part of the public API (which is TBD), it probably shouldn't be showing up in warning messages.

Reply:
In this specific case, we can keep the warning at the PeriodIndex level IMO. Setting the freq on PeriodArray can simply be disallowed, what we will do in the future for PeriodIndex as well?

Reply:
This was deprecated in 0.23 (https://github.com/pandas-dev/pandas/pull/20772) so we'll have to allow setting PeriodArray.freq for at least a version.

Reply:
But it maybe doesn't need to be through the public property? You can set `_freq` here, but already disallow setting the public `freq` on the array